src/bot/requestReview.ts

```diff
@@ -0,0 +1,123 @@
+import { CallbackParam, ShortcutParam } from '@/slackTypes';
+import { languageRepo } from '@repos/languageRepo';
+import { App, View } from '@slack/bolt';
+import log from '@utils/log';
+import { codeBlock, compose } from '@utils/text';
+import { BOT_ICON_URL, BOT_USERNAME } from './constants';
+import { ActionId, Deadline, Interaction } from './enums';
+
+export const requestReview = {
+  app: (undefined as unknown) as App,
+
+  setup(app: App): void {
+    log.d('requestReview.setup', 'Setting up RequestReview command');
+    this.app = app;
+    app.shortcut(Interaction.SHORTCUT_REQUEST_REVIEW, this.shortcut.bind(this));
+    app.view(Interaction.SUBMIT_REQUEST_REVIEW, this.callback.bind(this));
+  },
+
+  dialog(languages: string[]): View {
+    return {
+      title: {
+        text: 'Request a Review',
+        type: 'plain_text',
+      },
+      type: 'modal',
+      callback_id: Interaction.SUBMIT_REQUEST_REVIEW,
+      blocks: [
+        {
+          type: 'input',
+          label: {
+            text: 'What languages were used?',
+            type: 'plain_text',
+          },
+          element: {
+            type: 'checkboxes',
+            action_id: ActionId.LANGUAGE_SELECTIONS,
+            options: languages.map(language => ({
+              text: { text: language, type: 'plain_text' as const },
+              value: language,
+            })),
+          },
+        },
+        {
+          type: 'input',
+          label: {
+            text: 'When do you need this reviewed by?',
+            type: 'plain_text',
+          },
+          element: {
+            type: 'static_select',
+            action_id: ActionId.REVIEW_DEADLINE,
+            options: [
+              { text: { text: 'End of day', type: 'plain_text' }, value: Deadline.END_OF_DAY },
+              { text: { text: 'Tomorrow', type: 'plain_text' }, value: Deadline.TOMORROW },
+              { text: { text: 'End of week', type: 'plain_text' }, value: Deadline.END_OF_WEEK },
+              { text: { text: 'Monday', type: 'plain_text' }, value: Deadline.MONDAY },
+              { text: { text: 'Other', type: 'plain_text' }, value: Deadline.NONE },
```

Comment:
Would it be possible to have a text input when other is selected?

Reply:
Nope, did a bit of searching and I couldn't find any kind of conditional elements

Reply:
From the docs, it looked like there are events that are fired when interacting with blocks. I think you maybe can listen to those events and then update the dialog.

That seems to be how we can handle the button press for the accept / decline message